lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingService.java

```diff
@@ -259,29 +260,40 @@ public JourneyResponse checkVcResponse(
         }
 
         if (configService.enabled(DL_AUTH_SOURCE_CHECK)
-                && getDcmawDlVc(newVcs).isPresent()
-                && getSuccessfulDlVc(sessionCredentials).isEmpty()) {
+                && requiresAuthoritativeSourceCheck(newVcs, sessionVcs)) {
             return JOURNEY_DL_AUTH_SOURCE_CHECK;
         }
 
         return JOURNEY_NEXT;
     }
 
-    private Optional<VerifiableCredential> getDcmawDlVc(List<VerifiableCredential> vcs) {
+    private boolean requiresAuthoritativeSourceCheck(
+            List<VerifiableCredential> newVcs, List<VerifiableCredential> sessionVcs) {
+        return extractDrivingPermitIdentifier(newVcs, Cri.DCMAW, false)
+                .map(
+                        identifier ->
+                                !identifier.equals(
+                                        extractDrivingPermitIdentifier(
+                                                        sessionVcs, Cri.DRIVING_LICENCE, true)
+                                                .orElse(null)))
+                .orElse(false);
+    }
+
+    private Optional<String> extractDrivingPermitIdentifier(
+            List<VerifiableCredential> vcs, Cri cri, boolean successfulCheck) {
         return vcs.stream()
-                .filter(vc -> Cri.DCMAW.equals(vc.getCri()))
-                .filter(
+                .filter(vc -> cri.equals(vc.getCri()))
+                .filter(successfulCheck ? VcHelper::isSuccessfulVc : x -> true)
+                .map(
                         vc ->
                                 vc.getCredential().getCredentialSubject()
                                                 instanceof PersonWithDocuments personWithDocuments
-                                        && personWithDocuments.getDrivingPermit() != null)
-                .findFirst();
-    }
-
-    private Optional<VerifiableCredential> getSuccessfulDlVc(List<VerifiableCredential> vcs) {
-        return vcs.stream()
-                .filter(vc -> Cri.DRIVING_LICENCE.equals(vc.getCri()))
-                .filter(VcHelper::isSuccessfulVc)
+                                        ? personWithDocuments.getDrivingPermit()
+                                        : null)
+                .filter(Objects::nonNull)
+                .filter(drivingPermitDetailsList -> !drivingPermitDetailsList.isEmpty())
+                .map(drivingPermitDetailsList -> drivingPermitDetailsList.get(0))
+                .map(DrivingPermitDetails::getPersonalNumber)
```

Comment:
I wonder if we want to use the SPOT/CIMIT logic for this - https://govukverify.atlassian.net/wiki/spaces/Architecture/pages/3889922273/RFC+0042+Unique+identifiers+for+evidence+documents+for+de-duplicating+contra-indicators+in+Identity+Proofing+and+Verification+IPV#Driving-permits

That'll include the issuer etc., not just the number

(Up to you whether you want to implement the logic for _other_ document types)